modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java

```diff
@@ -250,6 +251,7 @@ public int hashCode() {
      * @return The name of the WKT element type, which is {@code "PROJCS"}.
      */
     @Override
+    @SuppressFBWarnings("ES_COMPARING_STRINGS_WITH_EQ")
```

Comment:
Wouldn't it be better to change x==y to x.equals(y) to fix the warning even if == is okay here instead of suppressing the warning?

Reply:
Using ``equals`` would be a performance difference; not sure how performance sensitive this is.

Reply:
I left it with == since the original author went to the trouble of writing a comment explaining that identity comparison was ok in that case. Also, code is convoluted (name is assigned during a if and result passed into a method as a side effect... horrible...)